apps/coscrad-frontend/src/components/higher-order-components/categorizables-of-multiple-type.container.tsx

```diff
@@ -1,29 +1,48 @@
-import { CategorizableType } from '@coscrad/api-interfaces';
+import { CategorizableCompositeIdentifier, CategorizableType } from '@coscrad/api-interfaces';
 import { buildPluralLabelsMapForCategorizableTypes } from '../../store/slices/resources/shared/connected-resources/build-plural-labels-map-for-categorizable-types';
 import { ICategorizableDetailPresenterFactory } from '../resources/factories/categorizable-detail-presenter-factory.interface';
 import { SelectedCategorizablesOfSingleTypeContainer } from './selected-categorizables-of-single-type.container';
 
-type CategorizableTypeAndSelectedIds = { [K in CategorizableType]?: string[] };
-
 interface CategorizablesOfMultipleTypeContainerProps<T> {
-    categorizableTypeAndIds: CategorizableTypeAndSelectedIds;
+    members: CategorizableCompositeIdentifier[];
     detailPresenterFactory: ICategorizableDetailPresenterFactory<T>;
     heading?: string;
 }
 
+const buildInitialEmptyReosurceTypesAndSelectedIds = (
+    members: CategorizableCompositeIdentifier[]
+) =>
+    [...new Set(members.map(({ type }) => type))].reduce(
+        (acc, resourceType) => ({
+            ...acc,
+            [resourceType]: [],
+        }),
+        {}
+    );
+
+// A Map might improve readability here
+const collectResourceTypesAndSelectedIds = (members: CategorizableCompositeIdentifier[]) =>
+    members.reduce(
+        (acc, { type: resourceType, id }) => ({
+            ...acc,
+            [resourceType]: acc[resourceType].concat(id),
+        }),
+        buildInitialEmptyReosurceTypesAndSelectedIds(members)
```

Comment:
May need your help to get this

Reply:
This is a common pattern when building maps (or hash maps using objects) where the values are arrays. You can either
- initialize a map with all the necessary keys pointing to empty arrays
- check every single time you encounter a key whether you need to initialize an empty array first
    - then push the new item
   
I find the initialize up front approach makes the subsequent reducer easier to parse.